### DISCUSSION TITLE
Rendered segment text across blocks

### DISCUSSION THREAD

**Discussion by piersmyers** [JanDeDobbeleer/oh-my-posh#3984] [General]
Hi Jan,

I am trying to use the rendered output of a segment but across blocks, so for example if I have one block that has a git segment and I want to use the rendered output of that segment in a different block, how do I do this (if it is at all possible)? It seems that I can only reference segments that are in the same block.

The reason I would like to do this is that currently I have a theme that still uses POSH_GIT for my console_title_template but I would like to replace that with the (rendered) text from my oh-my-posh git segment.

Thanks.

**Comment by JanDeDobbeleer** [comment]
@piersmyers been a while since I looked at this and my mind is blurry so I'll double check. I know for sure cross segment properties are sequential.

**Reply by JanDeDobbeleer** [reply]
@piersmyers so, you can't use the rendered output of a segment elsewhere today. That functionality doesn't exist. It also comes with a few challenges as template text isn't a plain string necessarily. It's potentially raw with color information (internal API). So this isn't possible I'm afraid.

**Reply by piersmyers** [reply]
Ah, that's a shame. Is it something you may think of adding in the future? The colour information (escape codes) could be stripped out - I think you already do this for the 'console_title_template' as the POSH_GIT variable has embedded formatting.

**Reply by JanDeDobbeleer** [reply]
@piersmyers I did indeed, but in this case these might not even be that. I can play around with it.

**Reply by JanDeDobbeleer** [reply]
@piersmyers this is what I mean, I can add the feature, but it's pretty useless:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/2492783/1804d0a7-c646-45c1-8a62-4e4ee1c80cc4)
